src/net_processing.cpp

```diff
@@ -2579,8 +2579,23 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
             pfrom.fDisconnect = true;
             return;
         }
-        // ignore this for now - later we can downgrade the resources allocated
-        // to this peer.
+        if (pfrom.IsFullOutboundConn() || pfrom.IsAddrFetchConn()) {
+            // If we picked an outbound for tx- or addr-relay and it sends us a
```

Comment:
I wonder if the rational shouldn't be different. AFAICT, BLOCKRELAY aims to inform the connected node about the initiator expectations on traffic relayed on this link. So a connected node inverting the logic and announcing its link expectations will break the effect and force the initiator to adapt its peer resource allocation.

I would say the restriction here is motivated only by the outboundness of the connection, the traffic class (tx, addr, block-relay) shouldn't matter. 

I think this PR and BIP introduce a "one-way" negotiation privilege for the initiator, which makes sense, but should this be documented in the BIP or stay an implicit assumption of our peer management policy ?